include/aws/common/linked_list.h

```diff
@@ -212,6 +212,34 @@ AWS_STATIC_IMPL void aws_linked_list_insert_after(
     AWS_POSTCONDITION(after->next == to_add);
 }
 
+/**
+ * Swaps the order two nodes in the linked list.
+ */
+AWS_STATIC_IMPL void aws_linked_list_swap_nodes(struct aws_linked_list_node *a, struct aws_linked_list_node *b) {
```

Comment:
It would be comforting to have unit tests for happy case,  a == b, head (of list, not the .head member) tail (same) cases, as well as separated (non-adjacent) if applicable